lib/engine/action/buy_shares.rb

```diff
@@ -5,13 +5,15 @@
 module Engine
   module Action
     class BuyShares < Base
-      attr_reader :entity, :bundle, :swap
+      attr_reader :entity, :bundle, :swap, :purchase_for, :borrow_from
```

Comment:
can you explain this mechanism?

Reply:
Ability to buy a share for another player and borrowing money from another player.

The `purchase_for` means it's player A's action, but they are buying a share for player B if you set `purchase_for` to player B. It's used for the union bank in 1871.

The `borrow_from` means if the player that's buying doesn't have enough money you can borrow cash from another player. Also used for the union bank in 1871.